lib/oidc/getWithPopup.ts

```diff
@@ -33,3 +33,47 @@ export function getWithPopup(sdk: OktaAuthOAuthInterface, options: TokenParams):
   });
   return getToken(sdk, options);
 }
+
+export function getWithIDPPopup(sdk: OktaAuthOAuthInterface, options: TokenParams): { cancel: () => void, promise: Promise<TokenResponse> } {
+  if (!BroadcastChannel) {
+    throw new AuthSdkError('Modern browser with `BroadcastChannel` support is required to use this method');
+  }
+
+  if (!options.state) {
+    options.state = generateState();
+  }
+
+  // some browsers (safari, firefox) block popup if it's initialed from an async process
+  // here we create the popup window immediately after user interaction
+  // then redirect to the /authorize endpoint when the requestUrl is available
+  const popupWindow = loadPopup('/', options);
+  const channel = new BroadcastChannel(`popup-callback:${options.state}`);
+
+  options = clone(options) || {};
+  Object.assign(options, {
+    display: 'popup',
+    responseMode: 'query',
+    popupWindow,
+    idpPopup: true,
+    channel,
+  });
+
+  let cancelPromise;
+  const promise = new Promise<TokenResponse>((resolve, reject) => {
+    cancelPromise = reject;
+    return getToken(sdk, options)
+    .then((res) => {
+      return resolve(res);
+    });
+  });
+
+  const cancel = () => {
+    channel.close();
```

Comment:
If the calling window calls `cancel`, should the popup window also be closed?

Reply:
Ideally yes, however that isn't possible when the `COOP` header is present, since the browser context shared between the main window and popup is disconnected